### PR TITLE
apiserver add --lease-reuse-duration-seconds to config lease reuse duration

### PR DESCRIPTION
**Pull request by lingsamuel**
Signed-off-by: Ling Samuel <lingsamuelgrace@gmail.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Details in #96836

**Which issue(s) this PR fixes**:

Part of #96836

**Special notes for your reviewer**:

Do we need to config `leaseReuseDurationPercent` too?

**Does this PR introduce a user-facing change?**:
```release-note
Add flag --lease-reuse-duration-seconds for kube-apiserver to config etcd lease reuse duration.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/lease_manager.go [hunk 1]
```diff
@@ -42,8 +42,8 @@ type leaseManager struct {
 }
 
 // newDefaultLeaseManager creates a new lease manager using default setting.
-func newDefaultLeaseManager(client *clientv3.Client) *leaseManager {
-	return newLeaseManager(client, 60, 0.05)
+func newDefaultLeaseManager(client *clientv3.Client, leaseReuseDurationSeconds int64) *leaseManager {
+	return newLeaseManager(client, leaseReuseDurationSeconds, 0.05)
```
Comment (lingsamuel) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/lease_manager.go:46]: We could change this to an option struct if we can config a max size of a lease further.

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd.go [hunk 2]
```diff
@@ -183,6 +183,9 @@ func (s *EtcdOptions) AddFlags(fs *pflag.FlagSet) {
 
 	fs.DurationVar(&s.StorageConfig.HealthcheckTimeout, "etcd-healthcheck-timeout", s.StorageConfig.HealthcheckTimeout,
 		"The timeout to use when checking etcd health.")
+
+	fs.Int64Var(&s.StorageConfig.LeaseReuseDurationSeconds, "etcd-lease-reuse-duration", s.StorageConfig.LeaseReuseDurationSeconds,
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go:187]: I was wondering about switching to DurationVar (same as above).

That would be cleaner, but given that etcd only supports second-granularity of leases (and that is reflected in storage API, etc.), maybe that is actually fine.
It seems we recently used this pattern too in: https://github.com/kubernetes/kubernetes/pull/95533


So let's leave this Int64Var, but change the flag to:

"lease-reuse-duration-seconds"

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/lease_manager.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/lease_manager.go:46]: Let's do that when we will be adding the next argument.

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd.go [hunk 3]
```diff
@@ -183,6 +183,9 @@ func (s *EtcdOptions) AddFlags(fs *pflag.FlagSet) {
...
+		"The time in seconds that each lease is reused. A lower value could avoid large number of objects reusing the same lease. Notice that a too small value may cause etcd performance problem.")
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go:188]: ... may cause performance problems at storage layer.

**Review by wojtek-t — COMMENTED** [review]
Thanks!

Just some minor comments.

/ok-to-test

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd.go (hunk 3 above)
Comment (lingsamuel) [reply staging/src/k8s.io/apiserver/pkg/server/options/etcd.go]: done

**Comment by wojtek-t** [comment]
@lingsamuel - please fix bazel (you need to autogenerate them) and I will approve once tests are green.

**Comment by mborsz** [comment]
/lgtm
Thanks!

**Pull request merged**